feat!: implement inflating tail emission

CjS77 opened the pull request:
## Summary

Adds a feature `tari_feature_mainnet_emission` to allow for tail emission
inflation. See #6122 

This change necessitates the addition of 2 new consensus constants:
`inflation_bips` -- the annual inflation rate of the total supply.
and `tail_emission_epoch_length`, which controls the tail emission
inflation. These replace `tail_emission`.

We update the Protobuf definition for `ConsensusConstants` to account for
the new fields.

Note: Replaces part of #6131

SWvheerden commented on applications/minotari_app_grpc/proto/types.proto:
```diff
@@ -119,7 +119,7 @@ message ConsensusConstants {
     uint64 emission_initial = 10;
     repeated uint64 emission_decay = 11;
-    uint64 emission_tail = 12;
+    uint64 emission_tail = 12 [deprecated=true];
```
Nit:
I think we can just delete this, it should default to 0 if you ask for it in a dated proto file. 
But I am 99% nothing uses this field

CjS77 replied:
Let's nuke it in a separate PR so that it's easy to revert if you're wrong.

SWvheerden reviewed (dismissed):
This code reads and follows so much easier without the feature flag in here.

CjS77 committed:
feat! inflating tail emission

Adds a feature `tari_feature_mainnet_emission` to allow for tail emission
inflation. See #6122

This change necessitates the addition of 2 new consensus constants:
`inflation_bips` -- the annual inflation rate of the total supply in
basis points (100 bips = 1 percentage point).
and `tail_emission_epoch_length`, which controls the tail emission
inflation.

These replace `tail_emission`.

We update the Protobuf definition for `ConsensusConstants` to account for
the new fields.

Updates tests

Adds `as_u128` to `MicroMinotari` since we need to perform a large
multiplication when calculating the inflation issuance.

Note: Replaces part of #6131

CjS77 committed:
fix lints
